src/vos/tests/evt_ctl.c

```diff
@@ -1255,18 +1385,26 @@ test_evt_iter_delete_internal(void **state)
 			assert_int_equal(rc, 0);
 		}
 	}
+	for (iter_count = 0; iter_count < (
+		sizeof(val)/sizeof(int)); iter_count++) {
+		/* No filter so delete everything */
+		rc = evt_iter_prepare(toh, val[iter_count], NULL, &ih);
+		assert_int_equal(rc, 0);
 
-	/* No filter so delete everything */
-	rc = evt_iter_prepare(toh, 0, NULL, &ih);
-	assert_int_equal(rc, 0);
-
-	rc = evt_iter_probe(ih, EVT_ITER_FIRST, NULL, NULL);
-	assert_int_equal(rc, 0);
-
-	/* Ok, delete the rest */
-	while (!evt_iter_empty(ih)) {
-		rc = evt_iter_delete(ih, NULL);
+		rc = evt_iter_probe(ih, EVT_ITER_FIRST, NULL, NULL);
 		assert_int_equal(rc, 0);
+
+		/* Ok, delete the rest */
+		while (!evt_iter_empty(ih)) {
+			rc = evt_iter_delete(ih, NULL);
+			if (val[iter_count] == 0) {
+				assert_int_equal(rc, 0);
+			} else {
+				assert_int_not_equal(rc, 0);
+				/* exit the loop */
+				break;
+			}
+		}
 	}
 	rc = evt_iter_finish(ih);
```

Comment:
This is a leak.   The evt_iter_finish should be in the same loop as evt_iter_prepare

Reply:
Done.